unity-renderer/Assets/DCLServices/WearablesCatalogService/WearablesCatalogService.cs

```diff
@@ -0,0 +1,97 @@
+﻿using Cysharp.Threading.Tasks;
+using System.Collections.Generic;
+using System.Threading;
+
+namespace DCLServices.WearablesCatalogService
+{
+    /// <summary>
+    /// This service will choose LambdasWearablesCatalogService or WebInterfaceWearablesCatalogService for requesting
+    /// wearables depending on the flag 'usingUrlParamsForDebug' received in the kernel configuration.
+    /// This will be temporal while we have to live with both services.
+    /// </summary>
+    public class WearablesCatalogService : IWearablesCatalogService
+    {
+        public BaseDictionary<string, WearableItem> WearablesCatalog =>
+            wearablesCatalogServiceInUse.WearablesCatalog;
+
+        private IWearablesCatalogService wearablesCatalogServiceInUse;
+        private readonly IWearablesCatalogService lambdasWearablesCatalogService;
+        private readonly IWearablesCatalogService webInterfaceWearablesCatalogService;
+        private readonly BaseDictionary<string, WearableItem> wearablesCatalog;
+        private readonly KernelConfig kernelConfig;
+
+        public WearablesCatalogService(
+            IWearablesCatalogService lambdasWearablesCatalogService,
+            IWearablesCatalogService webInterfaceWearablesCatalogService,
+            BaseDictionary<string, WearableItem> wearablesCatalog,
+            KernelConfig kernelConfig)
+        {
+            this.lambdasWearablesCatalogService = lambdasWearablesCatalogService;
+            this.webInterfaceWearablesCatalogService = webInterfaceWearablesCatalogService;
+            this.wearablesCatalog = wearablesCatalog;
+            this.kernelConfig = kernelConfig;
+        }
+
+        public void Initialize()
+        {
+            kernelConfig.EnsureConfigInitialized().Then(config => OnKernelConfigChanged(config, null));
```

Comment:
1. I think making `EnsureConfigInitialized` async is something that is out of the scope of this PR as it is being used also by other systems. I would move it to an independent `tech debt` task.
2. I decided to listen only one time to `kernelConfig.OnChange` because we never thought of making this "proxy" as something that could change of service dynamically in real time. Once we receive the kernel config at the beginning, the proxy choose the service to use and that's it. Why keep the subscription in this case?

Reply:
>I think making EnsureConfigInitialized async is something that is out of the scope of this PR as it is being used also by other systems. I would move it to an independent tech debt task.

You can already await it as it is a `Promise` and inherits `CustomYieldInstruciton`, you don't need to modify anything out of the scope of the task

>I decided to listen only one time to kernelConfig.OnChange because we never thought of making this "proxy" as something that could change of service dynamically in real time. Once we receive the kernel config at the beginning, the proxy choose the service to use and that's it. Why keep the subscription in this case?

Currently, you will call `OnKernelConfigChanged` twice: from the promise and from `OnChange` so the question is why it is called twice, not once?

Reply:
> You can already await it as it is a Promise and inherits CustomYieldInstruciton, you don't need to modify anything out of the scope of the task

Ok, I will do it!

> Currently, you will call OnKernelConfigChanged twice: from the promise and from OnChange so the question is why it is called twice, not once?

Oh, I already know what did you mean... I didn't realize that I didn't need to subscribe to the `OnChange` event if I was already using `EnsureConfigInitialized` 👍 I will change it!